chrome/src/background.ts

```diff
@@ -75,72 +75,105 @@ chrome.webNavigation.onBeforeNavigate.addListener(
 
 chrome.webRequest.onHeadersReceived.addListener(
   (info) => {
-    if (info.responseHeaders) {
-      const setCookieHeaders = info.responseHeaders
-        .filter(
-          (h): h is RequireFields<chrome.webRequest.HttpHeaders[0], 'value'> =>
-            h.name.toLowerCase() === 'set-cookie' &&
-            typeof h.value === 'string',
-        )
-        .map((h) => h.value)
-      if (setCookieHeaders.length === 0) {
-        return
-      }
-      setCookies({
-        setCookieHeaders,
-        requestUrl: info.url,
-        setCookie: async (cookie, domain) => {
-          await chrome.cookies.set(cookie)
-
-          const domainCookies = await chrome.cookies.getAll({
-            domain,
-          })
-
-          if (domainCookies.length > 0) {
-            await updateSessionRules({ domainCookies, RULE_ID })
-          }
-        },
-        removeCookie: async (cookie, domain) => {
-          await chrome.cookies.remove(cookie)
-
-          const domainCookies = await chrome.cookies.getAll({
-            domain,
-          })
+    if (info.parentFrameId < 0) {
+      return
+    }
 
-          if (domainCookies.length > 0) {
-            await updateSessionRules({ domainCookies, RULE_ID })
-          } else {
-            await chrome.declarativeNetRequest.updateSessionRules({
-              removeRuleIds: [RULE_ID],
-            })
+    // check the parent frame so we only override cookies if we are on nordcraft.com
+    chrome.webNavigation
+      .getFrame({
+        documentId: info.parentDocumentId,
+        frameId: info.parentFrameId,
+      })
+      .then((parentFrame) => {
+        if (!parentFrame) {
+          return undefined
+        }
+
+        const parentUrl = new URL(parentFrame.url)
+        if (
+          parentUrl.host.endsWith('toddle.dev') === false &&
+          parentUrl.host.endsWith('nordcraft.com') === false
+        ) {
+          return undefined
+        }
```

Comment:
This logic seems to be a duplicate of the code on line 17->36? Consider moving this to a separate function to make it easier to maintain - especially when (soon) need to remove 'toddle.dev' as a valid domain for the parent frame.